### PR TITLE
[8th May 2024] Add deprecation notices to CodeQL for VS Code documentation

### PR DESCRIPTION
**Pull request by subatoi**
On the 8th May, we're set to start automatically redirecting users from pages under https://codeql.github.com/docs/codeql-for-visual-studio-code/ to https://docs.github.com/en/code-security/codeql-for-vs-code/

This PR adds a deprecation note (reusable) to each of the pages under https://codeql.github.com/docs/codeql-for-visual-studio-code/, for those users whose browsers don't redirect. It also updates other instances of soon-to-be-outdated links in this repo, and adds a note to `CONTRIBUTING.md` about how to contribute to the newly migrated docs.

### CODE REVIEW

File: docs/codeql/reusables/vs-code-deprecation-note.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.. pull-quote::
+
+    Note
+
+    This documentation was migrated to docs.github.com/en/code-security/codeql-for-vs-code on 2024/05/08, and this version is no longer maintained.
```
Comment (jf205) [inline docs/codeql/reusables/vs-code-deprecation-note.rst:5]: Perhaps link to the changelog post?

Comment (subatoi) [reply docs/codeql/reusables/vs-code-deprecation-note.rst:5]: Good point 👍 

```suggestion
    This documentation `was migrated <https://github.blog/changelog/2024-04-18-codeql-for-visual-studio-code-documentation-is-now-on-docs-github-com/>`_ to docs.github.com/en/code-security/codeql-for-vs-code on 2024/05/08, and this version is no longer maintained.
```

**Review by mbg — DISMISSED** [review]
The changes in the `go/` directory look fine.

File: python/ql/src/CHANGELOG.md [hunk 2]
```diff
@@ -277,7 +277,7 @@ No user-facing changes.
 
 ### Bug Fixes
 
-* The [View AST functionality](https://codeql.github.com/docs/codeql-for-visual-studio-code/exploring-the-structure-of-your-source-code/) no longer prints detailed information about regular expressions, greatly improving performance.
+* The [View AST functionality](https://docs.github.com/en/code-security/codeql-for-vs-code/) no longer prints detailed information about regular expressions, greatly improving performance.
```
Comment (felicitymay) [inline python/ql/src/CHANGELOG.md:280]:
```suggestion
* The [View AST functionality](https://docs.github.com/en/code-security/codeql-for-vs-code/using-the-advanced-functionality-of-the-codeql-for-vs-code-extension/exploring-the-structure-of-your-source-code) no longer prints detailed information about regular expressions, greatly improving performance.
```

**Review by felicitymay — DISMISSED** [review]
Quite a few comments, but most are to link people to [Installing CodeQL for Visual Studio Code - GitHub Docs](https://docs.github.com/en/code-security/codeql-for-vs-code/getting-started-with-codeql-for-vs-code/installing-codeql-for-vs-code) when we tell them to set up the extension, rather than the generic article.

File: docs/codeql/reusables/vs-code-deprecation-note.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.. pull-quote::
+
+    Note
+
+    This documentation `was migrated <https://github.blog/changelog/2024-04-18-codeql-for-visual-studio-code-documentation-is-now-on-docs-github-com/>`_ to docs.github.com/en/code-security/codeql-for-vs-code on 2024/05/08, and this version is no longer maintained.
```
Comment (felicitymay) [inline docs/codeql/reusables/vs-code-deprecation-note.rst:5]:
```suggestion
    This documentation `was migrated <https://github.blog/changelog/2024-04-18-codeql-for-visual-studio-code-documentation-is-now-on-docs-github-com/>`__ to docs.github.com/en/code-security/codeql-for-vs-code on 2024/05/08, and this version is no longer maintained.
```

**Review by tausbn — DISMISSED** [review]
Python changes LGTM. 👍

**Review by felicitymay — APPROVED** [review]
I made a couple of small fixes, but this looks ready to merge.

**Comment by subatoi** [comment]
@jf205 👋 Could you please have another look at this one? It looks like after you're OK with it, we should be set to merge tomorrow 😄

File: README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ This open source repository contains the standard CodeQL libraries and queries t
 
 ## How do I learn CodeQL and run queries?
 
-There is [extensive documentation](https://codeql.github.com/docs/) on getting started with writing CodeQL using the [CodeQL extension for Visual Studio Code](https://codeql.github.com/docs/codeql-for-visual-studio-code/) and the [CodeQL CLI](https://codeql.github.com/docs/codeql-cli/).
+There is [extensive documentation](https://codeql.github.com/docs/) on getting started with writing CodeQL using the [CodeQL extension for Visual Studio Code](https://docs.github.com/en/code-security/codeql-for-vs-code/) and the [CodeQL CLI](https://docs.github.com/en/code-security/codeql-cli).
```
Comment (jf205) [inline README.md:7]: Tiny change to make it clearer that codeql.github.com/docs is now primarily the home for CodeQL language information.

```suggestion
There is extensive documentation about the [CodeQL language](https://codeql.github.com/docs/), writing CodeQL using the [CodeQL extension for Visual Studio Code](https://docs.github.com/en/code-security/codeql-for-vs-code/) and using the [CodeQL CLI](https://docs.github.com/en/code-security/codeql-cli).
```

**Review by jf205 — DISMISSED** [review]

**Comment by subatoi** [comment]
Thanks @jf205 @felicitymay if one of you could kindly reapprove, I'll aim to have this merged in sync with the redirects tomorrow

**Review by felicitymay — APPROVED** [review]

**Comment by subatoi** [comment]
Just to note that @saritai will be keeping an eye out for the [redirects PR](https://github.com/github/pages-lua/pull/570) to ship today. The person merging it is on Pacific Time and Sarita's on Eastern Time, so it will likely happen in the evening during British Summer Time.

cc @jf205

**Pull request merged**